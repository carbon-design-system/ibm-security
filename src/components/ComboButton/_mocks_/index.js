/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
