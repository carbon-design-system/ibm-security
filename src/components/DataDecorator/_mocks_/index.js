/**
 * @file Data decorator mocks.
 * @copyright IBM Security 2019
 */

const props = {
  type: 'IP',
  value: '10.0.0.0',
  score: 0,
  href: '#',
};
const midLine = {
  type: 'MD5',
  value: '0f3deda483df5e5f8043ea20297d243b',
  score: 30,
  href: '#',
};

export default props;

export { props, midLine };
