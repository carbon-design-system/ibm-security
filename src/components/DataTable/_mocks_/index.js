/* eslint-disable react/prop-types */
/**
 * @file Data table mocks.
 * @copyright IBM Security 2019
 */

import React from 'react';
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableHeader,
  TableRow,
  TableSelectAll,
  TableSelectRow,
} from '../../..';

const headers = [
  {
    key: 'lastSighted',
    header: 'Last sighted',
  },
  {
    key: 'name',
    header: 'Name',
  },
  {
    key: 'type',
    header: 'Type',
  },
  {
    key: 'attackPhase',
    header: 'Attack phase',
  },
  {
    key: 'method',
    header: 'Method',
  },
  {
    key: 'missing',
    header: 'Missing',
  },
];

const isSelectable = false;
const isSortable = false;

const rows = [
  {
    id: '7c34f4507156b35',
    lastSighted: '11/28/2018',
    name: 'IBM X-Force',
    type: 'IPv6',
    attackPhase: 'Initial Compromise',
    method: 'Phishing',
  },
  {
    id: '38347816a51b0c5a8',
    lastSighted: '06/28/2017',
    name: 'AppX',
    type: 'Hash',
    attackPhase: 'Establish Foothold',
    method: 'Deploy backdoor',
  },
  {
    id: '1d6116a2d1d03b4',
    lastSighted: '05/30/2018',
    name: 'Bigfix',
    type: 'URL',
    attackPhase: 'Escalate Privileges',
    method: 'Credential dumping',
  },
  {
    id: 'c52b0419981221765',
    lastSighted: '12/12/2018',
    name: 'Am I Affected',
    type: 'Hash',
    attackPhase: 'Cross-Site Scripting',
    method: 'Malicious Code',
  },
  {
    id: '697626e6f703131d',
    lastSighted: '11/11/2018',
    name: 'QRadar',
    type: 'Hash',
    attackPhase: 'SQL Injection Attack',
    method: 'SQL Vulnerabilities',
  },
  {
    id: 'd568f12fa7b7cac1',
    lastSighted: '12/09/2018',
    name: 'iris',
    type: 'Hash',
    attackPhase: 'Volume Based Attacks',
    method: 'ICMP floods',
  },
  {
    id: '697626e6f70313554',
    lastSighted: '12/10/2018',
    name: 'TISP',
    type: 'Hash',
    attackPhase: 'Application Layer Attacks',
    method: 'GET/POST floods',
  },
  {
    id: '1dd568f12fa7b7cac2',
    lastSighted: '12/11/2018',
    name: 'CLX',
    type: 'Hash',
    attackPhase: 'Protocol Attacks',
    method: 'SYN Floods',
  },
  {
    id: '697626e6f703131',
    lastSighted: '10/12/2018',
    name: 'X-Force Plus',
    type: 'Hash',
    attackPhase: 'Malicious Links',
    method: 'Social Engineering',
  },
  {
    id: 'dd568f12fa7b7cac0',
    lastSighted: '12/16/2018',
    name: 'BigFix & iris',
    type: 'Hash',
    attackPhase: 'Exploit Kits',
    method: 'Scripts',
  },
  {
    id: '697626e6f70313',
    lastSighted: '01/12/2018',
    name: 'Cases',
    type: 'Hash',
    attackPhase: 'Malicious Email',
    method: 'Social Engineering',
  },
];

const page = 1;
const pageSize = 5;
const pageSizes = [5, 10, 20];

/**
 * Render method.
 * @returns {string} The string to render.
 */
const render = ({
  rows,
  headers,
  getHeaderProps,
  getRowProps,
  getTableProps,
  getSelectionProps,
}) => (
  <TableContainer>
    <Table {...getTableProps()}>
      {headers && (
        <TableHead>
          <TableRow>
            <TableSelectAll {...getSelectionProps()} />
            {headers.map(header => (
              <TableHeader
                key={header.id}
                {...getHeaderProps({ header, isSortable })}
              >
                {header.header}
              </TableHeader>
            ))}
          </TableRow>
        </TableHead>
      )}
      <TableBody>
        {rows.map(row => (
          <TableRow {...getRowProps({ row })} key={row.id}>
            <TableSelectRow {...getSelectionProps({ row })} />
            {row.cells.map(cell => (
              <TableCell key={cell.id}>
                {cell.value ? cell.value : 'â€“'}
              </TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
    </Table>
  </TableContainer>
);

export {
  headers,
  isSelectable,
  isSortable,
  page,
  pageSize,
  pageSizes,
  render,
  rows,
};
