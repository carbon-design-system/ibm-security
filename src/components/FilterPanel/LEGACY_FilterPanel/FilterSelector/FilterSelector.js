/**
 * @file Filter selector.
 * @copyright IBM Security 2019
 */

import PropTypes from 'prop-types';
import React from 'react';

import { getComponentNamespace } from '../../../../globals/namespace';

import { filterFiltersPropTypes } from '../FilterPanelUtilities';
import FilterPanelCheckbox from '../../FilterPanelCheckbox';

export const namespace = getComponentNamespace('filter-selector');

const FilterSelector = ({ filter, onChange }) => (
  <FilterPanelCheckbox
    checked={filter.enabled}
    onChange={() => onChange(filter)}
    id={`filter-selector-${filter.id}`}
    count={filter.count}
    labelText={filter.name}
  />
);

FilterSelector.propTypes = {
  /** @type {Filter} Filter to render selector for */
  filter: filterFiltersPropTypes.isRequired,

  /** @type {(filter: Filter) => {}} Callback function when any filter is toggled */
  onChange: PropTypes.func,
};

FilterSelector.defaultProps = {
  onChange: () => {},
};

export default FilterSelector;
