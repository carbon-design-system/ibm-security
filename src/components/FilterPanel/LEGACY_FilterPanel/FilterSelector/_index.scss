////
/// Filter selector styles.
/// @group filter-panel
/// @copyright IBM Security 2019
////

@import '../../../Component/mixins';

@import 'mixins';

@include security--component($name: filter-selector) {
  @include filter-selector;
}
