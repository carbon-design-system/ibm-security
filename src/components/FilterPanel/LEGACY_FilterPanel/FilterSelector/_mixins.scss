////
/// Filter selector mixins.
/// @group filter-panel
/// @copyright IBM Security 2020
////

@import '../../FilterPanelCheckbox/mixins';

@mixin filter-selector {
  @include filter-panel-checkbox;
}
