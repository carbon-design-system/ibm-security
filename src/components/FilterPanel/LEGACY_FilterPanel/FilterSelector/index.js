/**
 * @file Filter Selector entry point.
 * @copyright IBM Security 2019
 */

export default from './FilterSelector';
