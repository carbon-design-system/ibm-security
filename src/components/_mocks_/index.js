/**
 * @file Global mocks.
 * @copyright IBM Security 2018
 */

import React from 'react';

import random from '../../globals/random';

const label = 'Label';

const lorem =
  'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiI+ICA8cGF0aCBmaWxsPSIjZmZmIiBkPSJNMTYgMThINmEyIDIgMCAwIDEtMi0yVjZhMiAyIDAgMCAxIDItMmgxMGEyIDIgMCAwIDEgMiAydjEwYTIgMiAwIDAgMS0yIDJ6TTYgNnYxMGgxMFY2em0yMCA2djRoLTR2LTRoNG0wLTJoLTRhMiAyIDAgMCAwLTIgMnY0YTIgMiAwIDAgMCAyIDJoNGEyIDIgMCAwIDAgMi0ydi00YTIgMiAwIDAgMC0yLTJ6bTAgMTJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnptLTEwIDJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnoiLz48L3N2Zz4=';

const id = '0';

const disabled = true;

const invalid = true;

const invalidText = 'Invalid text';

/**
 * Returns a child component.
 */
const children = <span>{label}</span>;

const className = 'class';

/**
 * Creates a URL for mock data.
 * @param {string} [hash=''] - A hash to pass to the URL method.
 * @returns {string} The mock data URL.
 */
const url = (hash = '') => `#${hash}`;

/**
 * Creates a `href` for mock data.
 * @returns {string} The `href` to return.
 */
const href = () => url();

export {
  children,
  className,
  disabled,
  href,
  icon,
  id,
  invalid,
  invalidText,
  label,
  lorem,
  random,
  url,
};
