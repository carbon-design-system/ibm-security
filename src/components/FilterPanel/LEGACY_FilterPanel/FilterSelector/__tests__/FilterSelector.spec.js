/**
 * @file Filter Selector tests.
 * @copyright IBM Security 2019
 */

import React from 'react';
import { shallow } from 'enzyme';
import FilterSelector from '..';
import { MockFilterData } from '../../../_mocks_';

describe('FilterSelector', () => {
  let filterSelector;

  beforeEach(() => {
    filterSelector = shallow(
      <FilterSelector
        filter={MockFilterData.filters.DEFAULT}
        onChange={() => {}}
      />
    );
  });

  describe('Rendering', () => {
    it('renders unchecked filter', () =>
      expect(filterSelector).toMatchSnapshot());

    it('renders checked filter', () => {
      filterSelector.setProps({ filter: MockFilterData.filters.SELECTED });
      expect(filterSelector).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('invokes `onChange` when filter is clicked', () => {
      const onChange = jest.fn();
      filterSelector.setProps({ onChange });
      filterSelector.find('FilterPanelCheckbox').simulate('change');
      expect(onChange).toHaveBeenCalledWith(MockFilterData.filters.DEFAULT);
    });
  });
});
