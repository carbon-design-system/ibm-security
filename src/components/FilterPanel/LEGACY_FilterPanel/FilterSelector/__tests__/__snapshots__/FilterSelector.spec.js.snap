// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSelector Rendering renders checked filter 1`] = `
<FilterCheckbox
  checked={true}
  count={10}
  id="filter-selector-SELECTED"
  indeterminate={false}
  labelText="Selected filter"
  onChange={[Function]}
/>
`;

exports[`FilterSelector Rendering renders unchecked filter 1`] = `
<FilterCheckbox
  checked={false}
  count={10}
  id="filter-selector-DEFAULT"
  indeterminate={false}
  labelText="Filter"
  onChange={[Function]}
/>
`;
